src/storage.js

```diff
@@ -114,9 +114,71 @@ const storageService = (provider, connectionString) => {
     };
   };
 
+  const makeAwsStorage = () => {
+    const AWS = require('aws-sdk');
+    const s3 = new AWS.S3();
+
+    const createWriteStream = (bucketName, fileName) => {
+      throw new Error("createWriteStream is not supported for AWS");
+    };
+
+    const getSignedUrl = (bucketName, fileName) => {
+      return new Promise((resolve, reject) => {
+        resolve(s3.getSignedUrl('getObject', {
+          Bucket: bucketName,
+          Key: fileName,
+          Expires: 60 * 5
+        }))
+      });
+    };
+
+    const downloadFileAsJson = (bucketName, fileName) => {
+      return new Promise((resolve, reject) => {
+        try {
+          s3.getObject({
+            Bucket: bucketName,
+            Key: fileName
+          }, (err, data) => {
+            if (err) return reject(err);
+            resolve(JSON.parse(data.Body.toString()))
+          });
+        } catch (e) {
+          reject("File is not in JSON format");
+        }
+      });
+    };
+
+
+    const uploadJsonBlob = (container, blobName, payload) => {
+      return new Promise((resolve, reject) => {
+        try {
```

Comment:
save here, try/catch is misplaced

Reply:
This one is correct. the parsing is done before the calling the API